snippets/zoe/src/offer-up.contract.js

```diff
@@ -0,0 +1,97 @@
+// #region file
+/** @file Contract to mint and sell a few Item NFTs at a time. */
+// @ts-check
+
+import { Far } from '@endo/far';
+import { M, getCopyBagEntries } from '@endo/patterns';
+import { AssetKind } from '@agoric/ertp/src/amountMath.js';
+import { atomicRearrange } from '@agoric/zoe/src/contractSupport/atomicTransfer.js';
+import '@agoric/zoe/exported.js';
+
+const { Fail, quote: q } = assert;
+
+// #region bagUtils
+/** @type { (xs: bigint[]) => bigint } */
+const sum = xs => xs.reduce((acc, x) => acc + x, 0n);
+
+/**
+ * @param {import('@endo/patterns').CopyBag} bag
+ * @returns {bigint[]}
+ */
+const bagCounts = bag => {
+  const entries = getCopyBagEntries(bag);
+  return entries.map(([_k, ct]) => ct);
+};
+// #endregion bagUtils
+
+/**
+ * In addition to the standard `issuers` and `brands` terms,
+ * this contract is parameterized by terms for price and,
+ * optionally, a maximum number of items sold for that price (default: 3).
+ *
+ * @typedef {{
```

Comment:
I wonder why GitHub actions is issuing a warning here:

```
Syntax error in type: {tradePrice: Amount;maxItems?: bigint;}
```

I wonder if some `tsc` version is out of date somewhere.

I'm not too worried about it just now, though.